File: JavaArchitecture/app/src/main/java/com/demo/architecture/injection/component/ActivityComponent.java
package com.demo.architecture.injection.component;

import com.demo.architecture.base.BaseActivity;
import com.demo.architecture.injection.PerActivity;
import com.demo.architecture.injection.module.ActivityModule;
import com.demo.architecture.ui.main.MainActivity;

import dagger.Subcomponent;

/**
 * Created by HeYingXin on 2018/4/17.
 */
@PerActivity
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(BaseActivity baseActivity);
    void inject(MainActivity mainActivity);
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/injection/component/ActivityComponent.kt
package com.mazouri.mvpkotlin.injection.component

import com.mazouri.mvpkotlin.base.BaseActivity
import com.mazouri.mvpkotlin.injection.PerActivity
import com.mazouri.mvpkotlin.injection.module.ActivityModule
import com.mazouri.mvpkotlin.task.main.MainActivity
import dagger.Subcomponent

/**
 * Created by HeYingXin on 2018/4/17.
 */
@PerActivity
@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

    fun inject(baseActivity: BaseActivity)
    fun inject(mainActivity: MainActivity)
}